spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb

```diff
@@ -1,5 +1,96 @@
 describe ManageIQ::Providers::Openstack::CloudManager::Refresher do
+  include Spec::Support::EmsRefreshHelper
+
+  before(:each) do
+    host = Rails.application.secrets.openstack[:hostname]
+    port = Rails.application.secrets.openstack[:port]
+    username = Rails.application.secrets.openstack[:userid]
+    password = Rails.application.secrets.openstack[:password]
+
+    zone = EvmSpecHelper.local_miq_server.zone
+    @ems = FactoryBot.create(:ems_openstack, :zone => zone, :hostname => host,
+                              :ipaddress => host, :port => port, :api_version => 'v3',
+                              :security_protocol => 'ssl-no-validation', :uid_ems => "default")
+
+    credentials = {:userid => username, :password => password}
+    @ems.authentications << FactoryBot.create(:authentication, credentials)
+  end
+
   it ".ems_type" do
     expect(described_class.ems_type).to eq(:openstack)
   end
+
+  it "will perform a full refresh" do
+    inventory = nil
+    2.times do # Run twice to verify that a second run with existing data does not change anything
+      with_vcr { @ems.refresh }
```

Comment:
We should refresh the network_manager and two storage managers here also:
```suggestion
with_vcr do
  @ems.refresh
  @ems.network_manager.refresh
  @ems.cinder_manager.refresh
  @ems.swift_manager.refresh
end
```